static/js/components/ValidationAlert.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import _ from 'lodash';
+
+import type { ValidationErrors } from '../flow/profileTypes';
+
+export default class ValidationAlert extends React.Component {
+  props: {
+    errors:   ValidationErrors,
+    children: React$Element[],
+  };
+
+  alertMessage: Function = (): React$Element|void => {
+    const { errors } = this.props;
+    if ( !_.isEmpty(errors) ) {
+      return (
+        <span className="message">
+          Please fix the errors above and try again
+        </span>
+      );
+    }
+  };
+
+  render () {
+    const { children } = this.props;
+    return (
+      <div className="validation-alert">
+        { this.alertMessage() }
+        <div className="actions">
+          { children }
+        </div>
+      </div>
```

Comment:
i wasn't very clear there. `ValidatedDialog` as i'm describing would be a component, not a utility class, and it would replace `ValidationAlert` (while keeping some of its functionality). it would look something like this (forgive the pseudo-code):

```javascript
class ValidatedDialog extends React.Component {
	ERROR_MESSAGE: "error"
	
	dialogActions() {
		const { actions, errors } = this.props
		if errors
			return <div className='validation-error'>{this.ERROR_MESSAGE}<div className='actions'>{actions}</div></div>
		else
			return actions
	}
	
	dialogProps() {
		return dialogActions() + (this.props without 'actions')
	}
	
	render() {
		return <Dialog {...this.dialogProps()} />
	}
}
```

benefits:

1. no unnecessary `<div className="validation-alert">` wrapper element where it's not necessary
1. wrapping the `actions` property is a hack that we need to get around a library limitation. better to keep that in one place
1. simple code reuse/easier API. you need to add a component to wrap the `actions` property in 3 different places in this PR. my suggestion replaces a component instead of adding a new component